text/0000-lwc-signals.md

```diff
@@ -0,0 +1,372 @@
+
+## Transition from `@wire`
+
+### Background
+
+The `@wire` decorator allows for a declarative approach to data fetching and reactive property updates based on changing inputs. It is closely tied to the Salesforce data layer and service layer.
+
+### New Approach
+
+Under the new design, external stores or Signals would take over the responsibility of managing reactive data fetching and updates. These stores would follow a standard protocol, making them framework-agnostic.
+
+__Advantages__:
+
+* __Flexibility__: Developers can use any data store that adheres to the protocol, not just Salesforce-specific services.
+* __Consistency__: A consistent pattern for managing both local and external data simplifies the mental model for developers.
+
+### Migration Path
+
+#### Replacing a `@wire` property
+
+We can go from this:
+
+```javascript
+import { RecordWireAdapter } from "c/wireAdapters";
+
+export default class ExampleComponent extends LightningElement {
+    @api recordId;
+    @wire(RecordWireAdapter, { recordId: "$recordId" }) record;
+}
+```
+
+To this:
+
+```javascript
+import { RecordStore } from "c/stores";
+
+export default class ExampleComponent extends LightningElement {
+    @api recordId;
+    record = new RecordStore(this.$api.recordId);
+}
+```
+
+This seems simple enough. No major differences, the `$` continues to be a thing in the new model so the store can access the value of the public property and react to it by fetching new data, it is just a different placement for the `$`. In terms of the usage of the record iself, it is almost the same, except that now we need to use `.value` to access the actual record in the template, e.g.:
+
+```html
+<template>
+    <p>{record.value.Name}</p>
+</template>
+```
+
+#### Replacing a `@wire` method
+
+We can go from this:
+
+```javascript
+import { RecordWireAdapter } from "c/wireAdapters";
+
+export default class ExampleComponent extends LightningElement {
+    @api recordId;
+    @wire(RecordWireAdapter, { recordId: "$recordId" }) updateRecord(record) {
+      // do something with the record
+    };
+}
+```
+
+To this:
+
+```javascript
+import { RecordStore } from "c/stores";
+
+export default class ExampleComponent extends LightningElement {
+    @api recordId;
+    record = new RecordStore(this.$api.recordId);
+
+    updateRecord(record) {
+      // do something with the record
+    }
+
+    connectedCallback() {
+        this.recordUnsubscript = this.record.subscribe(record => this.updateRecord(record));
+    }
+
+    disconnectedCallback() {
+        this.recordUnsubscript();
+    }
+}
+```
+
+It is important to highlight that this is the less common method of using a wire adapter. With the new approach, it becomes a little bit more cumbersome, but it is still possible to do it, and it offers a lot more flexibility since you can now use any store that follows the protocol, not just Salesforce-specific services.
```

Comment:
Most of the internal code I see uses wire methods, not properties.  Do we have numbers on the breakdown?